sdk/src/process_instruction.rs

```diff
@@ -321,20 +322,38 @@ pub mod stable_log {
         ic_logger_msg!(logger, "Program log: {}", message);
     }
 
+    /// Emit a program log data.
+    ///
+    /// The general form is:
+    ///
+    /// ```notrust
+    /// "Program log data: <program-id> <binary-data-in-base64>+"
+    /// ```
+    ///
+    /// That is, any program-generated output is guaranteed to be prefixed by "Program log data: "
+    pub fn program_log_data(logger: &Rc<RefCell<dyn Logger>>, program_id: &Pubkey, data: &[&[u8]]) {
+        ic_logger_msg!(
+            logger,
+            "Program log data: {} {}",
+            program_id,
+            data.iter().map(base64::encode).join(" ")
+        );
+    }
+
     /// Log return data as from the program itself. This line will not be present if no return
     /// data was set, or if the return data was set to zero length.
     ///
     /// The general form is:
     ///
     /// ```notrust
-    /// "Program return data: <program-id> <program-generated-data-in-base64>"
+    /// "Program return: <program-id> <program-generated-data-in-base64>"
     /// ```
     ///
     /// That is, any program-generated output is guaranteed to be prefixed by "Program return data: "
-    pub fn program_return_data(logger: &Rc<RefCell<dyn Logger>>, program_id: &Pubkey, data: &[u8]) {
```

Comment:
Why rename?

Reply:
That was a change proposed by @mvines 

Now that we're calling it "program log data:" I don't think it makes much sense to change this either, so I've reverted this change.

Reply:
I've flip flopped on this. Now they are:

```
Program data: <base64 event data>
Program return: <base64 return data>
```